From 593d06b11ec84d4d9250006f5213487b02a3aa1e Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Mon, 22 Apr 2013 10:37:44 +0200
Subject: [PATCH 6/6] mount: don't pass option "auto" to kernel

Fixes the following error if "auto" is set in fs options:
| EXT3-fs (sda1): error: unrecognized mount option "auto" or missing value
| EXT4-fs (sda1): Unrecognized mount option "auto" or missing value
| XFS (sda1): unknown mount option [auto].

Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 util-linux/mount.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/util-linux/mount.c b/util-linux/mount.c
index ebccaf6..fd3217d 100644
--- a/util-linux/mount.c
+++ b/util-linux/mount.c
@@ -222,6 +222,7 @@ static const int32_t mount_options[] = {
 	IF_FEATURE_MOUNT_FSTAB(
 		/* "defaults" */ 0,
 		/* "quiet" 0 - do not filter out, vfat wants to see it */
+		/* "auto" */ 0,
 		/* "noauto" */ MOUNT_NOAUTO,
 		/* "nofail" */ 0,
 		/* "sw"     */ MOUNT_SWAP,
@@ -282,6 +283,7 @@ static const char mount_option_str[] =
 	IF_FEATURE_MOUNT_FSTAB(
 		"defaults\0"
 		// "quiet\0" - do not filter out, vfat wants to see it
+		"auto\0"
 		"noauto\0"
 		"nofail\0"
 		"sw\0"
-- 
1.8.1.2

